### ISSUE TITLE
Help needen for  74HC165 PCB

### ISSUE THREAD

**Issue by Rehsifmas** [FreeJoy-Team/FreeJoyWiki#84]
![PCB](https://github.com/user-attachments/assets/a1d75146-8d52-49ce-a5de-6319bf2c1c99)

Hi there,
I'm building a 3d printed helicopter collective for a flightsimulator. I bought the shift register PCB as show in the image. But can anybody help me out how to connect buttons to it and how to set it up. Because I tried with no succes

Thanks in advance

**Comment by Assamita81** [comment]
Hey, it's a bit late, but do you still need help with this?

**Comment by Rehsifmas** [comment]
Nope, solved it.
Thanks for asking

Op wo 12 mrt 2025 09:09 schreef Assamita81 ***@***.***>:

**Comment by Solhead** [comment]
Hi, how did you fix it? I'm having the same problem. Can you tell me how you set it up? Thanks.

**Comment by Assamita81** [comment]
> Hi, how did you fix it? I'm having the same problem. Can you tell me how you set it up? Thanks.

Hi @Solhead. Do you have problems connecting the buttons to the board? or the board to the bluepill?

**Comment by Solhead** [comment]
I'm having trouble connecting the board to the Blu-ray player; it doesn't recognize the keystrokes. The AliExpress seller gave me instructions that don't match the instructions on the Freejoy Wiki. 

![Image](https://github.com/user-attachments/assets/141421b5-4010-4e46-9ac2-04d06f522426)
![Image](https://github.com/user-attachments/assets/c8937439-5c2f-4b06-a426-72c6af683424)

**Comment by Assamita81** [comment]
I assume the above screens are what the seller sent you. How do you have it connected?
So essentially, you need to set 3 pins in Freejoy: Clock, Latch, and Data, and those seem to be labeled in the board. The only thing is that you need to connect the OUT part of the board to the Bluepill. In your pictures above you are indicating the IN side is connected to the board. That is wrong.

**Comment by Apachekhoo** [comment]
![Image](https://github.com/user-attachments/assets/cbb4f4f7-3b57-40f4-8bac-51e9b5c55db9)

Hi, reviving this topic i have similar issues. My question is do i need to ensure all the GND (marked in red) are connected? 

No matter what i tried the shift register doesnt work.

**Comment by Assamita81** [comment]
> ![Image](https://github.com/user-attachments/assets/cbb4f4f7-3b57-40f4-8bac-51e9b5c55db9)
> 
> Hi, reviving this topic i have similar issues. My question is do i need to ensure all the GND (marked in red) are connected?
> 
> No matter what i tried the shift register doesnt work.

Those are ground pins, they are already connected between them. You need to connect one leg of each button to gnd, and the other to any of the rest of the pins. Are you connecting the OUT of the shift register to the bluepill? Have you configured it in the SR tab in freejoy?

**Comment by Apachekhoo** [comment]
Yes i connect using OUT
Yes I only connected one GND, i tested all the ground pins circled in red are connected
I have set the buttons to 24 as pullup in the SR tab
It still doesnt work….

I connect it to A (A3,4 and 5) series on the board. Do i have to connect to B series instead?

**Comment by Apachekhoo** [comment]
Yes i did.

**Comment by Assamita81** [comment]
I don't know what else to say. It may be a faulty one. If you have an arduino board laying around, I'd do some basic tests with it.
Another reasonable explanation is that the bluepill itself is the faulty one. The market is flooded with cheap knock offs that don't work very well. In fact, it's very difficult to find original ones, so that's also a very plausible explanation, that the problem is not the shift register, but the bluepill

**Comment by Apachekhoo** [comment]
Ok any idea where to get genuine bluepill? I think i got fake ones from aliexpress.Sent from my iPhoneOn Jun 18, 2025, at 10:15 PM, Assamita81 ***@***.***> wrote:﻿Assamita81 left a comment (FreeJoy-Team/FreeJoyWiki#84)
I don't know what else to say. It may be a faulty one. If you have an arduino board laying around, I'd do some basic tests with it.
Another reasonable explanation is that the bluepill itself is the faulty one. The market is flooded with cheap knock offs that don't work very well. In fact, it's very difficult to find original ones, so that's also a very plausible explanation, that the problem is not the shift register, but the bluepill

**Comment by Assamita81** [comment]
I don't see anything wrong with it, tbh. I'd suggest either test the shift register with a basic arduino sketch and serial print to see if it's faulty, or test a different one if you have more. I see you are building a tedac, so I guess you have more

**Comment by Assamita81** [comment]
As I said before, I don't see anything wrong with it. Maybe it's just a faulty board. I haven't used these myself, but I've seen how they're designed. I designed my own boards and had them manufactured, because I wanted to have individual shift registers, instead of packs of 3. You can find individual ones too, but then you need to add the pull up or down resistors externally. The few times I've had issues with them, it was just some pad not making a good connection

**Comment by Rehsifmas** [comment]
Did you connect it to a 3.3 Volt connection? The shift rehister doesn't work on volt

**Comment by Assamita81** [comment]
> Did you connect it to a 3.3 Volt connection? The shift rehister doesn't work on 5 volt

Not true, they are rated for 2-6V. Though to be honest, it's worth the try.

**Comment by Assamita81** [comment]
Although now that I think of, although the chip itself is rated for that voltage range, depending on the resistors value they use, it could impact the readings, so again, it's worth the try
